### PR TITLE
feat: updated dockerfile and revived docker functionality

### LINKED ISSUE #297: Bug: Dockerfile size is too large
### What happened?

Dockerfile size needs to be reduced as it will also reduce build time helps in optimisation

### Add screenshots

_No response_

### What browsers are you seeing the problem on?

_No response_

### Record

- [X] I agree to follow this project's Code of Conduct
- [X] I want to work on this issue

### PR DESCRIPTION
**Pull request by Bhavya-55**
## Related Issue

fixes #297 

## Description

reduced size and also updated dockerfile and revived functionality.
There is missing API key when trying to collect page data for the /api/send once that configure this dockerfile will run absolutely fine . pls if anyone into backend check that 

## Summary by CodeRabbit

- **Refactor**
    - Enhanced application performance and startup time by optimizing the Dockerfile with a new base image, updated file copying commands, modified dependency installation commands, adjusted build command, and refined container start command.

### CODE REVIEW

**Comment by Sanchitbajaj02** [comment]
> There is missing API key when trying to collect page data for the /api/send once that configure this dockerfile will run absolutely fine . pls if anyone into backend check that

Can you tell me where I can provide the Resend api key in this dockerfile from the .env assuming when someone pull this image, he/she has that?

Also few things to do:
- AI code review has suggested some changes please do that
- Update the branch with master to merge it

File: Dockerfile [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
+COPY package.json .
+COPY postcss.config.js .
+COPY tailwind.config.js .
+COPY tsconfig.json .
+ COPY yarn.lock .
 
 # Install app dependencies
-COPY ["package.json", "yarn.lock*", "./"]
+ #RUN npm install 
 RUN yarn install
```
Comment (Sanchitbajaj02) [inline Dockerfile:12]: How would yarn be installed into the image?

File: Dockerfile
Comment (Sanchitbajaj02) [inline Dockerfile:1]: How does image be created from source of app? I think there might be a script for it.
(Note: I don't know docker much so I am mentioning this from my theoretical knowledge of it)

**Review by Sanchitbajaj02 — CHANGES_REQUESTED** [review]

**Comment by Sanchitbajaj02** [comment]
Hi @Bhavya-55 
If you didn't respond in next 24hrs, I will close this repo

File: Dockerfile (hunk 1 above)
Comment (Bhavyaa55) [reply Dockerfile:12]: @Sanchitbajaj02 yarn can be installed like npm. Both are packages  . Yarn is faster that why I did yarn install

**Comment by Bhavyaa55** [comment]
@Sanchitbajaj02  I will make the relevant changes and all that needs to be done by today. By the end of day I will raise pr .

**Comment by Sanchitbajaj02** [comment]
Hi @Bhavya-55 
why are we first copying the config files and then copying entire file again `. .` ?

**Comment by Bhavyaa55** [comment]
@Sanchitbajaj02 We copy the specific configuration files and package.json first to ensure they are cached by Docker's layer caching mechanism. It is structured in a such a way to minimize redundant files

**Review by Sanchitbajaj02 — APPROVED** [review]
LGTM

**Comment by Bhavyaa55** [comment]
> LGTM.
> 
> Update the PR with latest changes to get it merged

Thanks for reviewing changes but pr is already updated , what needs to be done further let me know . I have already pushed the recent changes ..

**Pull request merged**